tests/Meziantou.Analyzer.Test/Rules/OptimizeLinqUsageAnalyzerUseCastInsteadOfSelectTests.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System.Threading.Tasks;
+using Meziantou.Analyzer.Rules;
+using Xunit;
+using TestHelper;
+
+namespace Meziantou.Analyzer.Test.Rules
+{
+    public sealed class OptimizeLinqUsageAnalyzerUseCastInsteadOfSelectTests
+    {
+        private static ProjectBuilder CreateProjectBuilder()
+        {
+            return new ProjectBuilder()
+                .WithAnalyzer<OptimizeLinqUsageAnalyzer>(id: RuleIdentifiers.OptimizeEnumerable_CastInsteadOfSelect)
+                .WithCodeFixProvider<OptimizeLinqUsageFixer>();
+        }
+
+        [Theory]
+        [InlineData("enumerable.[|Select|](dt => (BaseType)dt)",
+                    "enumerable.Cast<BaseType>()")]
+        [InlineData("Enumerable.[|Select|](enumerable, dt => dt as Test.BaseType).Where(x => x != null)",
```

Comment:
You're right, my bad.

Modified the tests and adapted code so that using `as` operator will no longer trigger a diagnostic.

We could also decide to simply offer another fix for it: the use of `OfType`. What do you think?

Reply:
`OfType<T>()` is the equivalent of `Select(item => item as Test.BaseType).Where(x => x != null)`. This is more complex to detect, but if you want to, you can create another PR to support it.

Reply:
Then I guess it's out of scope for now.